React/Base/RCTConvert.m

```diff
@@ -1145,22 +1145,49 @@ + (NSPropertyList)NSPropertyList:(id)json
 #if TARGET_OS_OSX // [TODO(macOS ISS#2323203)
 + (NSString*)accessibilityRoleFromTrait:(NSString*)trait
 {
-  // a subset of iOS accessibilityTraits map to macOS accessiblityRoles:
-  if ([trait isEqualToString:@"button"]) {
-    return NSAccessibilityButtonRole;
-  } else if ([trait isEqualToString:@"text"]) {
-    return NSAccessibilityStaticTextRole;
-  } else if ([trait isEqualToString:@"link"]) {
-    return NSAccessibilityLinkRole;
-  } else if ([trait isEqualToString:@"image"]) {
-    return NSAccessibilityImageRole;
-    // a set of RN accessibilityTraits are macOS specific accessiblity roles:
-  }	else if ([trait isEqualToString:@"group"]) {
-    return NSAccessibilityGroupRole;
-  } else if ([trait isEqualToString:@"list"]) {
-    return NSAccessibilityListRole;
+  static NSDictionary<NSString *, NSString *> *traitOrRoleToAccessibilityRole;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    traitOrRoleToAccessibilityRole = @{
+      // from https://reactnative.dev/docs/accessibility#accessibilityrole
+      @"adjustable": NSAccessibilitySliderRole,
+      @"alert": NSAccessibilityStaticTextRole, // no exact match on macOS
+      @"button": NSAccessibilityButtonRole, // also a legacy iOS accessibilityTraits
+      @"checkbox": NSAccessibilityCheckBoxRole,
+      @"combobox": NSAccessibilityComboBoxRole,
+      @"header": NSAccessibilityStaticTextRole, // no exact match on macOS
+      @"image": NSAccessibilityImageRole, // also a legacy iOS accessibilityTraits
+      @"imagebutton": NSAccessibilityButtonRole, // no exact match on macOS
+      @"keyboardkey": NSAccessibilityButtonRole, // no exact match on macOS
+      @"link": NSAccessibilityLinkRole, // also a legacy iOS accessibilityTraits
+      @"menu": NSAccessibilityMenuRole,
+      @"menubar": NSAccessibilityMenuBarRole,
+      @"menuitem": NSAccessibilityMenuBarItemRole,
+      @"none": NSAccessibilityUnknownRole,
+      @"progressbar": NSAccessibilityProgressIndicatorRole,
+      @"radio": NSAccessibilityRadioButtonRole,
+      @"radiogroup": NSAccessibilityRadioGroupRole,
+      @"scrollbar": NSAccessibilityScrollBarRole,
+      @"search": NSAccessibilityStaticTextRole, // no exact match on macOS
```

Comment:
it won't be a text field?

Reply:
Agreed, makes more sense to map to NSAccessibilityTextFieldRole